### ISSUE TITLE
Bug: `no-unassigned-vars` incorrectly flags SolidJS refs

### ISSUE THREAD

**Issue by ulrichstark** [eslint/eslint#20169]
### Environment

Node version: v24.9.0
npm version: v11.6.0
Local ESLint version: v9.36.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin ### 24.6.0

### What parser are you using?

Default (Espree)

### What did you do?

```js
import { onMount } from "solid-js";

export function Input() {
    let inputRef: HTMLInputElement | undefined;
    onMount(() => inputRef?.select());
    return <input ref={inputRef} />;
}
```

### What did you expect to happen?

inputRef shouldn't be reported as never assigned

### What actually happened?

inputRef is incorrectly reported as never assigned

### Link to Minimal Reproducible Example

[ESLint Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHF1b3RlczogW1wiZXJyb3JcIiwgXCJkb3VibGVcIl0gKi9cblxuaW1wb3J0IHsgb25Nb3VudCB9IGZyb20gXCJzb2xpZC1qc1wiO1xuXG5leHBvcnQgZnVuY3Rpb24gSW5wdXQoKSB7XG4gICAgbGV0IGlucHV0UmVmOiBIVE1MSW5wdXRFbGVtZW50IHwgdW5kZWZpbmVkO1xuICAgIG9uTW91bnQoKCkgPT4gaW5wdXRSZWY/LnNlbGVjdCgpKTtcbiAgICByZXR1cm4gPGlucHV0IHJlZj17aW5wdXRSZWZ9IC8+O1xufSIsIm9wdGlvbnMiOnsicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOlsiZXJyb3IiXSwiZm9yLWRpcmVjdGlvbiI6WyJlcnJvciJdLCJnZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOlsiZXJyb3IiXSwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8tY2xhc3MtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbXBhcmUtbmVnLXplcm8iOlsiZXJyb3IiXSwibm8tY29uZC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3QtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWJpbmFyeS1leHByZXNzaW9uIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6WyJlcnJvciJdLCJuby1jb250cm9sLXJlZ2V4IjpbImVycm9yIl0sIm5vLWRlYnVnZ2VyIjpbImVycm9yIl0sIm5vLWRlbGV0ZS12YXIiOlsiZXJyb3IiXSwibm8tZHVwZS1hcmdzIjpbImVycm9yIl0sIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6WyJlcnJvciJdLCJuby1kdXBlLWVsc2UtaWYiOlsiZXJyb3IiXSwibm8tZHVwZS1rZXlzIjpbImVycm9yIl0sIm5vLWR1cGxpY2F0ZS1jYXNlIjpbImVycm9yIl0sIm5vLWVtcHR5IjpbImVycm9yIl0sIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1lbXB0eS1wYXR0ZXJuIjpbImVycm9yIl0sIm5vLWVtcHR5LXN0YXRpYy1ibG9jayI6WyJlcnJvciJdLCJuby1leC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjpbImVycm9yIl0sIm5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl0sIm5vLWZ1bmMtYXNzaWduIjpbImVycm9yIl0sIm5vLWdsb2JhbC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW1wb3J0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbnZhbGlkLXJlZ2V4cCI6WyJlcnJvciJdLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6WyJlcnJvciJdLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6WyJlcnJvciJdLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1uZXctbmF0aXZlLW5vbmNvbnN0cnVjdG9yIjpbImVycm9yIl0sIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjpbImVycm9yIl0sIm5vLW9iai1jYWxscyI6WyJlcnJvciJdLCJuby1vY3RhbCI6WyJlcnJvciJdLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOlsiZXJyb3IiXSwibm8tcmVkZWNsYXJlIjpbImVycm9yIl0sIm5vLXJlZ2V4LXNwYWNlcyI6WyJlcnJvciJdLCJuby1zZWxmLWFzc2lnbiI6WyJlcnJvciJdLCJuby1zZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjpbImVycm9yIl0sIm5vLXNwYXJzZS1hcnJheXMiOlsiZXJyb3IiXSwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC1wcml2YXRlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtY2F0Y2giOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXSwibm8td2l0aCI6WyJlcnJvciJdLCJyZXF1aXJlLXlpZWxkIjpbImVycm9yIl0sInVzZS1pc25hbiI6WyJlcnJvciJdLCJ2YWxpZC10eXBlb2YiOlsiZXJyb3IiXSwibm8tdW5hc3NpZ25lZC12YXJzIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfSwic291cmNlVHlwZSI6Im1vZHVsZSJ9LCJwYXJzZXIiOiJAdHlwZXNjcmlwdC1lc2xpbnQvcGFyc2VyIn19fQ==
)

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

SolidJS docs for refs: https://docs.solidjs.com/concepts/refs#refs-in-solid

**Comment by kecrily** [comment]
I don't believe this is a bug in the `no-unassigned-vars` rule. You can [use `eslint-disable` comment to disable it](https://eslint.org/docs/latest/use/configure/rules#disabling-rules) where needed.

```js
import { onMount } from "solid-js";

export function Input() {
    let inputRef; // eslint-disable-line no-unassigned-vars
    onMount(() => inputRef?.select());
    return <input ref={inputRef} />;
}
```

**Comment by jamesarosen** [comment]
It would be lovely to be able to configure the set of allowed unused vars, much in the way `no-unused-vars` supports `varsIgnorePattern: "^_"`. That way, we could do `varsIgnorePattern: "Ref$"` and not have to add a comment every time we use a ref in Solid.

**Comment by yoiang** [comment]
@kecrily sorry to bother you, can you please provide more explanation as to why this shouldn't be considered a bug? I may be dim, I don't think I understand the reasoning, considering from what I remember it does detect react-refs a-ok? 

Or how can we rectify the situation safely (which are some of the purpose of tools like type and assignment safety) without using `eslint-disable`?

Being explicit using the following then resolves into **Property 'inputRef' does not exist on type 'never'** reported by Typescript, would you say that the proper fix should be something clarified by Solidjs' typing with Typescript?
```ts
import { onMount } from "solid-js";

export function Input() {
    let inputRef: HTMLElement | undefined = undefined;
    onMount(() => inputRef?.select());
    return <input ref={inputRef} />;
}
```
